.github/workflows/mkosi.yml

```diff
@@ -75,6 +75,8 @@ jobs:
           - distro: centos
             release: "9"
             sanitizers: ""
+    env:
+      CFLAGS: -Werror
```

Comment:
I wonder if it would be possible to switch the Fedora ASan/UBSan job to clang and somehow pass `--verbose` to `ninja` there?

Reply:
With clang and ASan/UBSan `test-bus-vtable-cc` seems to be failing with
```
>>> UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 SYSTEMD_LANGUAGE_FALLBACK_MAP=/work/src/src/locale/language-fallback-map
 SYSTEMD_SLOW_TESTS=0 PATH=/work/src/obj-x86_64-linux-gnu:/usr/lib/linux-tools/6.8.0-31-generic:/usr/lib/linux-tools/6.8.0-31-generic:/work/scripts:/usr/bin:/
usr/sbin MALLOC_PERTURB_=246 PROJECT_BUILD_ROOT=/work/src/obj-x86_64-linux-gnu EFI_ADDON=/work/src/obj-x86_64-linux-gnu/src/boot/efi/addonx64.efi.stub LD_LIBR
ARY_PATH=/work/src/obj-x86_64-linux-gnu/src/shared /work/src/obj-x86_64-linux-gnu/test-bus-vtable-cc                                                          
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――                                                                                
stderr:                                                                                                                                                       
==344191==Your application is linked against incompatible ASan runtimes.                                                                                      
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― 
```
I'll try to pass `--verbose` to `ninja`

Reply:
Hmm I'm still seeing this, yet everything seems linked correctly:

```
daandemeyer-fedora-PC1EV17T:/work/src # ldd /work/build/test-bus-vtable-cc
        linux-vdso.so.1 (0x00007fff272fc000)
        libsystemd-shared-256.so => /work/build/src/shared/libsystemd-shared-256.so (0x00007f8305400000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f83051a8000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f8307b24000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f8307b11000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f830517b000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f8304f76000)
        libclang_rt.asan-x86_64.so => /usr/lib64/clang/18/lib/linux/libclang_rt.asan-x86_64.so (0x00007f830444a000)
        libacl.so.1 => /lib64/libacl.so.1 (0x00007f8307b08000)
        libblkid.so.1 => /lib64/libblkid.so.1 (0x00007f830440e000)
        libcap.so.2 => /lib64/libcap.so.2 (0x00007f8307afa000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f83043d2000)
        libmount.so.1 => /lib64/libmount.so.1 (0x00007f8304385000)
        libcrypto.so.3 => /lib64/libcrypto.so.3 (0x00007f8303e32000)
        libpam.so.0 => /lib64/libpam.so.0 (0x00007f8303e20000)
        libseccomp.so.2 => /lib64/libseccomp.so.2 (0x00007f8303e00000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f8303dd2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8307c13000)
        libeconf.so.0 => /lib64/libeconf.so.0 (0x00007f8303dc4000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f8303da1000)
        libaudit.so.1 => /lib64/libaudit.so.1 (0x00007f8303d76000)
        libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007f8303cca000)
daandemeyer-fedora-PC1EV17T:/work/src # /work/build/test-bus-vtable-cc
==1843145==Your application is linked against incompatible ASan runtimes.
daandemeyer-fedora-PC1EV17T:/work/src # ldd /lib64/libstdc++.so.6
        linux-vdso.so.1 (0x00007ffccdbda000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f06c3a5a000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f06c3855000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f06c3da1000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f06c3828000)
daandemeyer-fedora-PC1EV17T:/work/src # ldd /work/build/src/shared/libsystemd-shared-256.so
        linux-vdso.so.1 (0x00007ffebfb0f000)
        libclang_rt.asan-x86_64.so => /usr/lib64/clang/18/lib/linux/libclang_rt.asan-x86_64.so (0x00007f1aa58d4000)
        libacl.so.1 => /lib64/libacl.so.1 (0x00007f1aa859f000)
        libblkid.so.1 => /lib64/libblkid.so.1 (0x00007f1aa8563000)
        libcap.so.2 => /lib64/libcap.so.2 (0x00007f1aa8555000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f1aa8519000)
        libmount.so.1 => /lib64/libmount.so.1 (0x00007f1aa84cc000)
        libcrypto.so.3 => /lib64/libcrypto.so.3 (0x00007f1aa5381000)
        libpam.so.0 => /lib64/libpam.so.0 (0x00007f1aa84ba000)
        libseccomp.so.2 => /lib64/libseccomp.so.2 (0x00007f1aa849a000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f1aa846a000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f1aa529a000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f1aa5095000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f1aa4e3d000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f1aa843d000)
        libeconf.so.0 => /lib64/libeconf.so.0 (0x00007f1aa842f000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f1aa840a000)
        libaudit.so.1 => /lib64/libaudit.so.1 (0x00007f1aa4e12000)
        libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007f1aa4d66000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f1aa85b0000)
```

Reply:
I passed `--verbose` to `ninja` and `-shared-libasan` wasn't passed to `clang` as far as I can tell.

With the latest version of this PR `-shared-libasan` is passed and the test passes on my machine:
```sh
[2179/3100] clang++ -Itest-bus-vtable-cc.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wno-missing-field-initializers -Wno-unused-parameter -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/work/src=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection '-fdebug-prefix-map=/work/src=/usr/src/systemd-256~rc3-20240601230045' -O0 -shared-libasan -fno-sanitize=function -Wdate-time -D_FORTIFY_SOURCE=0 -fno-strict-aliasing -fstrict-flex-arrays=1 -fvisibility=hidden -ffunction-sections -fdata-sections -fno-omit-frame-pointer -include config.h -MD -MQ test-bus-vtable-cc.p/src_libsystemd_sd-bus_test-bus-vtable-cc.cc.o -MF test-bus-vtable-cc.p/src_libsystemd_sd-bus_test-bus-vtable-cc.cc.o.d -o test-bus-vtable-cc.p/src_libsystemd_sd-bus_test-bus-vtable-cc.cc.o -c ../src/libsystemd/sd-bus/test-bus-vtable-cc.cc
systemd/sd-bus/test-bus-vtable-cc.cc
[2180/3100] clang++  -o test-bus-vtable-cc test-bus-vtable-cc.p/src_libsystemd_sd-bus_test-bus-vtable-cc.cc.o -fsanitize=address,undefined -Wl,--as-needed -Wl,--no-undefined -fuse-ld=lld -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-rpath=/usr/lib/llvm-18/lib/clang/18/lib/linux -specs=/usr/share/debhelper/dh_package_notes/debian-package-notes.specs -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/work/src=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection '-fdebug-prefix-map=/work/src=/usr/src/systemd-256~rc3-20240601230045' -O0 -shared-libasan -fno-sanitize=function -Wdate-time -D_FORTIFY_SOURCE=0 '-Wl,-rpath,$ORIGIN/src/shared:XXXXXXXXXXXXXXX' -Wl,-rpath-link,/work/src/obj-x86_64-linux-gnu/src/shared -Wl,--start-group src/shared/libsystemd-shared-256.so -Wl,--end-group -Wl,--fatal-warnings -Wl,-z,now -Wl,-z,relro -Wl,--gc-sections
clang++: warning: argument unused during compilation: '-specs=/usr/share/debhelper/dh_package_notes/debian-package-notes.sp
```
(I'm not sure why I see `-D_FORTIFY_SOURCE=0` instead of `-U_FORTIFY_SOURCE` there though. The `argument unused during compilation` warning looks weird too)

Reply:
Regarding `-Werror` I think it would be better to pass `--werror` to `meson` instead. https://github.com/mesonbuild/meson/issues/7360 was fixed recently so the `-Werror` kludge in the other actions can probably be revisited. Another option would probably be to keep passing both `-Werror` and `--werror` just in case.

Reply:
@evverx The fix is too recent to be available on CentOS 9. So for now I'd prefer to do it via CFLAGS until a more recent meson is available in all the distributions we support.